runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -346,10 +348,7 @@ public static StreamingDataflowWorker fromOptions(DataflowWorkerHarnessOptions o
             .setSizeMb(options.getWorkerCacheMb())
             .setSupportMapViaMultimap(options.isEnableStreamingEngine())
             .build();
-    Function<String, ScheduledExecutorService> executorSupplier =
-        threadName ->
-            Executors.newSingleThreadScheduledExecutor(
```

Comment:
I think changing `Executors.newSingleThreadScheduledExecutor` to  `Executors.newSingleThreadExecutor` should fix the problem? We don't need to introduce a new Executor.

Reply:
We need `ScheduledExecutorService` here and `Executors.newSingleThreadExecutor` does not return a `ScheduledExecutorService`

Question here:

do we want active work refresh failures to crash the JVM? if yes than we need a terminating executor, if not i think we can remove `TerminatingExecutors`, use  `Executors.newSingleThreadExecutor` everywhere we want a failure to crash the JVM and `ScheduledExecutorService` everywhere we are ok with failures.

Reply:
Resolved https://github.com/apache/beam/pull/32922#discussion_r1820519685

Reply:
yeah, since that schedules at a fixed rate internally to ActiveWorkRefresher I don think we want to crash the vm.